### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by SomeoneWeird**
- Adds some authentication middleware to most routes
- Starts mocking out (with mostly-real routes [dogeorg/dkm](https://github.com/dogeorg/dkm)) a DKM manager, which just hits a child pup.

### TODO: 
- [ ] Figure out how we're going to get a pup manifest/state properly. We need this to understand where the container lives on the system (for its veth IP)
- [x] Get the frontend hooked up properly.

### CODE REVIEW

**Review by blendtwenty — APPROVED** [review]
Works effectively with frontend.

- Am rejected with incorrect pass 👍
- Am provided token on correct pass 👍
- Am rejected when trying to access protected apis with expired or invalid token 👍

File: pkg/manifest.go [hunk 1]
```diff
@@ -17,6 +17,7 @@ type PupManifest struct {
 	PermissionGroups []PermissionGroup `json:"permissionGroups"`
 	Dependencies     []Dependency      `json:"dependencies"`
 	hydrated         bool
+	containerIP      string
```
Comment (tjstebbing) [inline pkg/manifest.go:20]: The PupState for installed pups has the IP of the container

**Review by tjstebbing — APPROVED** [review]
👍

File: pkg/manifest.go (hunk 1 above)
Comment (SomeoneWeird) [reply pkg/manifest.go:20]: Yup, this was temporary until we landed some of your changes.

**Pull request merged**